pkg/hive/job/job.go

```diff
@@ -454,18 +481,30 @@ func (jo *jobObserver[T]) start(ctx context.Context, wg *sync.WaitGroup, options
 
 	if jo.fn == nil {
 		l.Error("can't start a nil observer job")
+		options.metrics.JobErrorsTotal.WithLabelValues(jo.name).Inc()
 		return
 	}
 
 	l.Debug("Observer job started")
 
 	done := make(chan struct{})
 
-	var err error
+	var (
+		stat = &spanstat.SpanStat{}
+		err  error
+	)
 	jo.observable.Observe(ctx, func(t T) {
+		stat.Start()
+
 		err := jo.fn(ctx, t)
+
+		sec := stat.End(true).Seconds()
+		options.metrics.ObserverRunDuration.WithLabelValues(jo.name).Observe(sec)
+		stat.Reset()
+
 		if err != nil {
 			l.WithError(err).Error("Observer job errored")
+			options.metrics.JobErrorsTotal.WithLabelValues(jo.name).Inc()
```

Comment:
btw, we're logging "context cancelled" errors when stopping here. I think it'd make sense to check `ctx.Err()` for cancellation and then not log the error (since it's expected behaviour when stopping).